src/parser/parser.yy

```diff
@@ -1682,7 +1682,11 @@ match_clause
         $$ = new MatchClause($2, $3, false/*optional*/);
     }
     | KW_OPTIONAL KW_MATCH match_path_list where_clause {
-        $$ = new MatchClause($3, $4, true);
+        if ($4 != nullptr) {
+            throw nebula::GraphParser::syntax_error(@4, "Where clause in optional match is not supported.");
```

Comment:
delete $3 before throwing error to avoid memory leak of $3.

Reply:
and delete $4

Reply:
Fixed.